staging/src/k8s.io/legacy-cloud-providers/azure/clients/securitygroupclient/azure_securitygroupclient_test.go

```diff
@@ -102,6 +217,201 @@ func TestList(t *testing.T) {
 	assert.Equal(t, 3, len(result))
 }
 
+func TestListNextResultsMultiPages(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	tests := []struct {
+		prepareErr error
+		sendErr    *retry.Error
+		statusCode int
+		body	   io.ReadCloser
+	}{
+		{
+			prepareErr: nil,
+			sendErr:    nil,
+			statusCode: http.StatusOK,
+		},
+		{
+			prepareErr: fmt.Errorf("error"),
+		},
+		{
+			sendErr: &retry.Error{RawError: fmt.Errorf("error")},
+			statusCode: http.StatusOK,
+		},
+	}
+
+	lastResult := network.SecurityGroupListResult{
+		NextLink: to.StringPtr("next"),
+	}
+
+	isRead := false
+	for _, test := range tests {
+		armClient := mockarmclient.NewMockInterface(ctrl)
+		req := &http.Request{
+			Method: "GET",
+		}
+		armClient.EXPECT().PrepareGetRequest(gomock.Any(), gomock.Any()).Return(req, test.prepareErr)
+		if test.prepareErr == nil {
+			armClient.EXPECT().Send(gomock.Any(), req).Return(&http.Response{
+				StatusCode: http.StatusOK,
+				Body:       ioutil.NopCloser(bytes.NewReader([]byte(`{"foo":"bar"}`))),
+			}, test.sendErr)
+			armClient.EXPECT().CloseResponse(gomock.Any(), gomock.Any())
+		}
+
+		sgClient := getTestSecurityGroupClient(armClient)
+		result, err := sgClient.listNextResults(context.TODO(), lastResult)
+		if test.prepareErr != nil || test.sendErr != nil {
+			assert.NotNil(t, err)
+		} else {
+			assert.NoError(t, err)
+		}
+		if test.prepareErr != nil {
+			assert.Empty(t, result)
+		} else {
+			reader1 := bytes.NewReader([]byte(`{"foo":"bar"}`))
+			if isRead == false {
```

Comment:
@feiskyer I have fixed all issues, except I meet problem here, so I have to hack reader offset using reader1.Seek, and err as following, is it a product bug or by design?
```
Error:          Not equal:
                                expected: network.SecurityGroupListResult{Response:autorest.Response{Response:(*http.Response)(0xc0000f9b90)}, Value:(*[]network.SecurityGroup)(nil), NextLink:(*string)(nil)}
                                actual  : network.SecurityGroupListResult{Response:autorest.Response{Response:(*http.Response)(0xc0000f9b00)}, Value:(*[]network.SecurityGroup)(nil), NextLink:(*string)(nil)}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -14,3 +14,3 @@
                                      },
                                -     i: (int64) 0,
                                +     i: (int64) 13,
                                      prevRune: (int) -1
```